static/js/stream_data.js

```diff
@@ -197,15 +197,16 @@ exports.render_stream_description = function (sub) {
 
 exports.update_calculated_fields = function (sub) {
     sub.is_admin = page_params.is_admin;
-    // Admin can change stream name/description either stream is public or
-    // stream is private and admin is subscribed to private stream.
-    sub.can_change_name_description = page_params.is_admin &&
-                                     (!sub.invite_only || (sub.invite_only && sub.subscribed));
+    // User can perform stream admin actions either user is subscribed stream admin
+    // OR user is org admin and stream public OR user is org admin subscriber of private stream.
+    sub.can_perform_admin_actions = (page_params.is_admin && (!sub.invite_only || sub.subscribed))
+                                    || (sub.is_stream_admin && sub.is_subscribed);
```

Comment:
Add one extra `can_perform_admin_actions` flag, which check whether user have the admin rights for stream or not. This flag can be use further in different admin actions flag.